erpnext/selling/doctype/sales_order/sales_order.js

```diff
@@ -98,10 +98,25 @@ frappe.ui.form.on("Sales Order", {
 	},
 
 	delivery_date: function(frm) {
+		if (!frm.doc.customer) {
+			frappe.throw(__('Please select a customer'));
```

Comment:
Empty the `delivery_date` field before throwing the error, otherwise it'll not re-trigger after selecting the customer.

Or we can add this trigger for the `customer` field too, maybe. But that might throw the error multiple times.